recipe_scrapers/_abstract.py

```diff
@@ -65,41 +66,41 @@ def host(cls) -> str:
         """get the host of the url, so we can use the correct scraper"""
         raise NotImplementedError("This should be implemented.")
 
-    def canonical_url(self):
+    def canonical_url(self) -> str | None:
         canonical_link = self.soup.find("link", {"rel": "canonical", "href": True})
         if canonical_link:
-            return urljoin(self.url, canonical_link["href"])
+            return urljoin(self.url, canonical_link["href"])  # type: ignore[type-var,return-value,index]
         return self.url
 
-    def title(self):
+    def title(self) -> str | None:
```

Comment:
Although the simplified type union syntax in Python 3.10 is nice, we [support Python 3.6 and above](https://github.com/hhursev/recipe-scrapers/blob/3ec02b540397be4fc21ce4f9e700aa0c730303d2/pyproject.toml#L22) at the moment.